kubetest/azure.go

```diff
@@ -537,7 +537,36 @@ func (c Cluster) Down() error {
 }
 
 func (c Cluster) DumpClusterLogs(localPath, gcsPath string) error {
-	return nil
+	// should run on master node and collect logs through winrm from windows slaves
+
+	const AzureLogDumpTemplate = `
+function wrapper-ssh () {
+	ssh -i %[1]s -o StrictHostKeyChecking=no "$@"
+}
+function wrapper-scp () {
+	scp -i %[1]s -o StrictHostKeyChecking=no "$@"
+}
+wget https://raw.githubusercontent.com/e2e-win/e2e-win-prow-deployment/master/custom-scripts/master.sh
+wrapper-ssh azureuser@%[2]s 'cat | bash /dev/stdin --user azureuser --pass Passw0rdAdmin --output /tmp/out' < master.sh
+wrapper-scp -r azureuser@%[2]s:/tmp/out/ %[3]s
+`
+	// Prevent an obvious injection.
+	if strings.Contains(localPath, "'") || strings.Contains(gcsPath, "'") {
+		return fmt.Errorf("%q or %q contain single quotes - nice try", localPath, gcsPath)
+	}
+
+	var masterHostname string
+	masterHostname = fmt.Sprintf("%s.%s.cloudapp.azure.com", c.dnsPrefix, c.location)
```

Comment:
Extract dns name template into a variable. Much cleaner when we need to change it for some reason.